File: iterator/OrderIterator.php
<?php
require_once __DIR__.'/BaseIterator.php';

class OrderIterator extends BaseIterator
{
    public function __construct($values)
    {
        $this->values = $values;
        $this->index = -1;
    }

    public function next()
    {
        return $this->values[++$this->index];
    }

    public function hasNext()
    {
        return isset($this->values[$this->index + 1]);
    }
}

File: iterator-advance/app/OrderIterator.java
class OrderIterator extends BaseIterator
{
    public OrderIterator(Item[] values)
    {
        this.values = values;
        this.index = -1;
    }

    public Item next()
    {
        return this.values[++this.index];
    }

    public boolean hasNext()
    {
        return (this.values.length-1) > this.index;
    }
}